src/cdp.jl

```diff
@@ -145,17 +148,15 @@ end
 #= Methods =#
 
 function _s_wise_max(cdp::ContinuousDP, s, C)
+    sp = Array{Float64}(size(cdp.shocks, 1), length(s))
     function objective(x)
-        out = 0.
-        t = Base.tail(indices(cdp.shocks))
-        for (i, w) in enumerate(cdp.weights)
-            e = cdp.shocks[(i, t...)...]
-            out += w * funeval(C, cdp.interp.basis, cdp.g(s, x, e))
+        for i in 1:size(sp, 1)
+            sp[i, :] = cdp.g(s, x, cdp.shocks[i, :])
```

Comment:
I was supposing the dimension of `cdp.shocks` may be larger than 2 (for example, the realization of shocks may be a matrix), but let's just assume that the shocks are scalar or vector (1de101f447c90c819bd7ec4a240a8cadbf5331b9).

Reply:
Ahh ok, good point. I think matrix should be sufficiently general as higher dimension arrays can always be reshaped into a matrix and will naturally be stacked as columns due to the column major array ordering